pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -3,6 +3,42 @@
 stdenv:
 
 let
+  inherit (lib)
```

Comment:
```suggestion
  # these attributes are inherited to the lexical scope for performance reasons
  inherit (lib)
```

Reply:
It's also more concise.

Reply:
Uh, this PR adds 118 lines.  I don't consider that concise.

Using `with` instead, to eliminate those 118 lines, would make the code significantly less-readable.  Also I don't think cppnix has tvix's optimization that makes non-nested `with` statements as efficient as `let inherit`.

Reply:
Indeed it does not. (and please just call it C++ Nix or whatever, even if that's still unnecessarily reductive in my opinion)

I suppose it behaves like a source transformation that goes from
```
with e1; e2
```
to
```
let inherit (e1) <free variables of e2>;
in e2
```

You could even generalize that for nested `with` using some `//`. I suppose tvix does not implement that because multiple `with` is even more frowned upon than a single `with`?

If you have more info, I'd be interested.

Reply:
> adds 118 lines. I don't consider that concise.
> 
> Using `with` instead, to eliminate those 118 lines, would make the code significantly less-readable.

These two sentences seem contradictory. I agree with the latter.
For me concision only matters locally, so I'm happy to ignore those lines.
We already have plenty of those in `lib/` and I'm never annoyed with them, because I don't see them when jumping between functions.
If this is a problem for you, please sort out a jump to definition solution and/or search integration in your editor. It's pretty much a prerequisite for being productive in any non-trivial codebase.